### PR TITLE
Cross-test with Scala 3

### PR DESCRIPTION
**Pull request by julienrf**
Remaining issues

- [x] Fix compilation of http4s-server tests. There seems to be an issue in the usage of the Akka-http graph DSL in Scala 3. We use it to convert between Akka-streams and http4s streams.
- [x] Compilation of openapi module loops indefinitely

Relates to #566

### CODE REVIEW

**Comment by julienrf** [comment]
Any help to move forward would be appreciated, I don’t have time to look at the remaining issues.

**Comment by ptrdom** [comment]
I do not have permissions to push to your branch, so here is commit to my fork that fixes http4s-server test compilation - https://github.com/ptrdom/endpoints4s/commit/d3e4f09a9d7785965d65233eff90b00d68c7438e.

**Comment by ptrdom** [comment]
And this fixes openapi tests - https://github.com/ptrdom/endpoints4s/commit/f3b2b4d9829fe4960e0698a756218186fe640905.

**Comment by ptrdom** [comment]
Remaining Scala 3 errors are all in example code, at a quick glance they do look like issues in build definition.

**Comment by ptrdom** [comment]
You will probably notice, but just a note that you might not want to take the following change from my commits -
https://github.com/ptrdom/endpoints4s/commit/d3e4f09a9d7785965d65233eff90b00d68c7438e#diff-993fc08217429dd4f90458a62bd3f0c461dd5c8c8031173e323d6fa94e95f21bR67-R70.

**Comment by julienrf** [comment]
Thank you very much @ptrdom!

> I do not have permissions to push to your branch, so here is commit to my fork that fixes http4s-server test compilation - [ptrdom@d3e4f09](https://github.com/ptrdom/endpoints4s/commit/d3e4f09a9d7785965d65233eff90b00d68c7438e).

IIUC we only need to bump akka and use `createGraph` instead of `create`? Nice!

> And this fixes openapi tests - [ptrdom@f3b2b4d](https://github.com/ptrdom/endpoints4s/commit/f3b2b4d9829fe4960e0698a756218186fe640905).

Interesting! How did you figure out the cause of the issue? I wonder if there is a bug in Scala 3 or if this is expected.

**Comment by ptrdom** [comment]
> IIUC we only need to bump akka and use createGraph instead of create? Nice!

Yeah, I noticed a comment about it in the code of the first Akka release for Scala 3 - figured it was very likely they ran into this issue too.

> Interesting! How did you figure out the cause of the issue? I wonder if there is a bug in Scala 3 or if this is expected.

I just noticed that the class in error had private implicits in inner class, changed that - did not seem to help, then I moved the inner class definitions out to the companion object and that did it.

I see we have more errors now, will check them out.

**Comment by julienrf** [comment]
Ah, about f3b2b4d, I think this is related to a limitation that has been fixed in lampepfl/dotty#15847, which has been shipped in Scala 3.2.0. We could also try to bump the Scala 3 version to 3.2.1, maybe?

**Comment by ptrdom** [comment]
> Ah, about [f3b2b4d](https://github.com/endpoints4s/endpoints4s/commit/f3b2b4d9829fe4960e0698a756218186fe640905), I think this is related to a limitation that has been fixed in [lampepfl/dotty#15847](https://github.com/lampepfl/dotty/pull/15847), which has been shipped in Scala 3.2.0. We could also try to bump the Scala 3 version to 3.2.1, maybe?

Neat, will try it out!

**Comment by ptrdom** [comment]
~We could also try to bump the Scala 3 version to 3.2.1, maybe?~ (actually, I am fine with your change)

All good. That change does improve the code quality, so to speak - those definitions do not benefit from being inside the class, so might as well be in the companion object.

File: akka-http/build.sbt [hunk 1]
```diff
@@ -55,7 +55,6 @@ val `akka-http-server` =
       excludeDependencies ++= {
         if (scalaBinaryVersion.value.startsWith("3")) {
           List(
-            ExclusionRule("org.scala-lang.modules", "scala-xml_3"),
```
Comment (ptrdom) [inline akka-http/build.sbt:58]: @julienrf Is this change fine? Tests will not work on Scala 3 with this exclusion.

**Review by ptrdom — CHANGES_REQUESTED** [review]

**Review by ptrdom — COMMENTED** [review]
Checks pass, just need to double check that exclusion change.

File: akka-http/build.sbt (hunk 1 above)
Comment (julienrf) [reply akka-http/build.sbt:58]: Yes, that’s even better! Eventually, we should get rid of those exclusions.

Comment (julienrf) [reply akka-http/build.sbt:58]: That makes me think that we should probably not publish Scala 3 artifacts for the modules that still use `CrossVersion.for3Use2_13` in their dependencies… :thinking:

**Comment by julienrf** [comment]
That’s amazing, thank you @ptrdom!

**Pull request merged**